app/scripts/modules/core/src/search/global/GlobalSearch.tsx

```diff
@@ -36,26 +39,53 @@ export class GlobalSearch extends React.Component<{}, IGlobalSearchState> {
   private searchField: HTMLInputElement;
   private resultRefs: HTMLElement[][];
 
+  private query$ = new Subject<string>();
+  private destroy$ = new Subject();
+
   constructor() {
     super();
     this.state = {
-      showSearchResults: false,
-      showRecentItems: false,
+      showDropdown: false,
       showMinLengthWarning: false,
       query: '',
       querying: false,
-      categories: []
+      categories: null
     };
   }
 
   public componentDidMount() {
     window.addEventListener('keyup', this.handleWindowKeyup);
     window.addEventListener('click', this.handleWindowClick);
+
+    const { infrastructureSearchService } = ReactInjector;
+    const search = infrastructureSearchService.getSearcher();
+
+    this.query$
+      .switchMap((query: string) => (
+        Observable.fromPromise(search.query(query))
+      ))
+      .map((result) => (
+        result
+          .filter(({ results }) => results.length)
+          .map((category) => ({
+            ...category,
+            results: searchRank(category.results).slice(0, 5)
+          }))
+          .sort((a, b) => a.type.order - b.type.order)
+      ))
+      .takeUntil(this.destroy$)
+      .subscribe((categories) => {
+        this.resultRefs = categories.map(() => []);
+
+        this.setState({ querying: false, categories });
+      });
   }
 
   public componentWillUnmount() {
     window.addEventListener('keyup', this.handleWindowKeyup);
```

Comment:
Shouldn't these be `removeEventListener`?

Reply:
🤦‍♂️ 🤦‍♂️ 🤦‍♂️ 

The even sadder thing is, I distinctly remember fixing this copy-pasta when I was working on the previous PR, and somehow it managed to still make it through on that one.

![disappointment](https://bukk.it/disappointment.jpg)